draft-ietf-sipcore-sip-push.xml

```diff
@@ -545,6 +545,86 @@ align="center"><artwork>
+        this information will not be present in mid-dialog requests towards the UA. Instead, the proxy 
+        need to support a mechanism where it stores the information needed to request a push notification 
+        towards the UA and be able to find and retrieve that information when it receives a mid-dialog 
+        request. This section defines such mechanism. The UA and proxy procedures in this section are 
+        applied in addition to the generic procedures defined in this specification.
+      </t>
+      <section anchor="section.subscription.ua" title="SIP UA Behavior">
+        <section anchor="section.subscription.ua.sub" title="Initial Request for Dialog">
+          <t>
+            When the UA sends in initial request for a dialog, or a 2xx response to such requests, if the UA is 
+            willing to receive push notifications triggered by incoming mid-dialog requests, the UA MUST include 
+            a 'pn-purk' SIP URI parameter in the Contact header of the request or response. The UA MUST include a 
+            parameter value identical to the the last 'sip.pnspurk' feature-capability indicator that it received 
+            in a REGISTER response.
+          </t>
+          <t>
+            The UA decision whether it is willing to receive push notifications triggered by incoming mid-dialog requests
+            is done based on local policy. Such policy might be based on the type of SIP dialog, the type of media (if any) 
+            negotiated for the dialog <xref target="RFC3264"/>, etc.
+          </t>
+        </section>
+      </section>  
+      <section anchor="section.subscription.proxy" title="SIP Proxy Behavior">
+        <section anchor="section.subscription.proxy.reg" title="REGISTER">
+          <t>
+            When the proxy receives an initial REGISTER request for a registration from the UA, if the proxy supports
+            requesting push notfifications, triggered by mid-dialog requests, towards the registered UA, the proxy MUST 
+            store the information (the pn- SIP URI parameters) needed to request push notifications associated with the registration 
+            towards the UA. In addition the proxy MUST generate a random value, referred to as the PURK (Proxy Unique Registration Key), 
+            that is unique within the context of the proxy and that can be used as a key to retrieve the information. When the proxy
+            receives the associated 2xx REGISTER response, it adds a 'sip.pnspurk' feature-capability indicator with the PURK value 
+            to the associcated 2xx REGISTER response. When the proxy receives a binding refresh REGISTER request, it MUST add a 'sip.pnspurk' 
+            feature-capability indicator with the previously generated key as the value to the associated 2xx REGISTER response.
+          </t>
+          <t>
+            The random value generated by the proxy MUST be generated in a way so that it cannot be used to retrieve information
```

Comment:
The **unique** value generated by the proxy MUST be generated in such a way so that it cannot be used to retrieve information about the user associate with registration for instance by utilizing a cyptographically secure random function.